functions/that-api-webhooks/hooks/receive/auth0SignUp.js

```diff
@@ -0,0 +1,182 @@
+const Sentry = require('@sentry/node');
+const debug = require('debug');
+const ac = require('../send/activeCampaign');
+
+const dlog = debug('that:api:webhooks:auth0SignUp');
+
+module.exports = (req, res, next) => {
+  dlog('trigger (receive) auth0SignUp');
+  let payload = [];
+  if (req.method === 'POST') {
+    dlog('post');
+    if (req.body) {
+      usePayload(req.body, res, next);
+    } else {
+      req
+        .on('error', err => {
+          console.error(err);
+        })
+        .on('data', chunk => {
+          dlog('data event');
+          payload.push(chunk);
+        })
+        .on('end', () => {
+          dlog('end event');
+          payload = Buffer.concat(payload).toString();
+          usePayload(payload, res, next);
+        });
+    }
+  } else {
+    console.error(`Non POST request, ${req.method}`);
+    Sentry.captureMessage(`Non POST request, ${req.method}`, 'info');
+    res.writeHead(405, { 'Content-Type': 'text/plain' });
+    res.write(`Unsupported method.`);
+    res.end();
+  }
+};
+
+async function usePayload(payload, res, next) {
+  dlog('use payload from auth0 %O', payload);
+  const SIGNUP_TAG = 'NewSignUp';
+  const REGFROM_FIELD_ID = '20';
+  const THATUS_SYSTEMS_MSG_LIST = 'THAT.us System Messages';
+  // end state goal is contact is added to AC automation "User Sign-up No Profile"
+  // see automation for trigger details
+
+  if (!payload.email) {
+    dlog('stopping. missing email in payload');
+    res.writeHead(400, { 'Content-type': 'text/plain' });
+    res.write(`Bad Request. Did you forget your payload?`);
+    res.end();
+    return;
+  }
+  let contactResult;
+  let tagResult;
+  let listResult;
+  try {
+    [contactResult, tagResult, listResult] = await Promise.all([
+      ac.findContactByEmail(payload.email),
+      ac.searchForTag(SIGNUP_TAG),
+      ac.searchForList(THATUS_SYSTEMS_MSG_LIST),
+    ]);
+  } catch (e) {
+    next(e);
+  }
+  dlog('contactResult %o', contactResult);
+  dlog('tagResult %o', tagResult);
+
+  if (!tagResult || (tagResult && !tagResult.id)) {
+    dlog(`tag, ${SIGNUP_TAG}, not found at AC`);
+    Sentry.captureMessage(`Failed to find tag at AC ${SIGNUP_TAG}`, 'error');
```

Comment:
just call next with the new error.

Reply:
corrected in all instances in file. Actually much cleaner now :)